quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -98,28 +98,28 @@ impl LocalFileStorage {
     }
 }
 
-/// Ensure that the path given does not include any ".." for security reasons.
-///
-/// In order to reduce the attack surface, we want to make sure the `FileStorage`
-/// only access/delete files that are children of its root_directory.
-fn ensure_valid_relative_path(path: &Path) -> StorageResult<()> {
-    for component in path.components() {
-        match component {
-            Component::RootDir | Component::ParentDir | Component::Prefix(_) => {
-                // We forbid `Path` components that are breaking the assumption that
-                // root.join(path) is a child of root (if we omit fs links).
-                return Err(StorageErrorKind::Unauthorized.with_error(anyhow::anyhow!(
-                    "path `{}` is forbidden. only simple relative path are allowed",
-                    path.display()
-                )));
-            }
-            Component::CurDir | Component::Normal(_) => {
-                // we accept `./` and subdir/
-            }
-        }
-    }
-    Ok(())
-}
+// /// Ensure that the path given does not include any ".." for security reasons.
+// ///
+// /// In order to reduce the attack surface, we want to make sure the `FileStorage`
+// /// only access/delete files that are children of its root_directory.
+// fn ensure_valid_relative_path(path: &Path) -> StorageResult<()> {
```

Comment:
same question as above?

Reply:
let's have only one discussion thread for this: https://github.com/quickwit-oss/quickwit/pull/3869#discussion_r1338768504